cms/static/js/views/modals/move_xblock_modal.js

```diff
@@ -101,6 +103,26 @@ function($, Backbone, _, gettext, BaseView, BaseModal, XBlockInfoModel, MoveXBlo
             );
         },
 
+        enableMoveOperation: function (isValidMove) {
+            var $moveButton = this.$el.find('.action-move');
+            if (isValidMove){
+                $moveButton.removeClass('is-disabled');
+            } else {
+                $moveButton.addClass('is-disabled');
+            }
+        },
+
+        validateMoveOperation: function (targetParentXBlockInfo) {
+            var isValidMove = false,
+                sourceParentType = this.sourceParentXBlockInfo.get('category'),
+                targetParentType = targetParentXBlockInfo.get('category');
+
+            if (targetParentType == sourceParentType && this.sourceParentXBlockInfo.id != targetParentXBlockInfo.id) {
+                isValidMove = true;
+            }
+            this.enableMoveOperation(isValidMove);
```

Comment:
Apparently, this method should only validate the move, doing enable/disable should't be job of this method. can we move this `this.enableMoveOperation(isValidMove);` out and return bool value.